### PR TITLE
fix: handle empty JULIAHUB_PROJECT_UUID var correctly

### PR DESCRIPTION
**Pull request by pfitzseb**

### CODE REVIEW

File: src/authentication.jl [hunk 1]
```diff
@@ -515,6 +515,8 @@ function _juliahub_project(
     elseif isa(project, UUIDs.UUID)
         return project
     elseif isa(project, AbstractString)
+        isempty(project) && return nothing
```
Comment (mortenpi) [inline src/authentication.jl:518]: I think it would be better to add this only for the actual environment variable above. I.e. if the user passes an explicit empty string, we should still error. But let's `isempty(strip(get(ENV, "JULIAHUB_PROJECT_UUID", ""))) -> nothing` the environment variable?

**Comment by mortenpi** [comment]
Let's also add tests for the edge cases in https://github.com/JuliaComputing/JuliaHub.jl/blob/main/test/authentication.jl ?
